cmake/CMakeLists.txt

```diff
@@ -128,6 +129,8 @@ option(onnxruntime_ENABLE_TRAINING "Enable training functionality." OFF)
 option(onnxruntime_ENABLE_TRAINING_E2E_TESTS "Enable training end-to-end tests." OFF)
 option(onnxruntime_USE_HOROVOD "Build with HOROVOD support" OFF)
 option(onnxruntime_USE_NCCL "Build with NCCL support" ON)
+#A special option just for debugging and sanitize check. Please do not enable in option in retail builds.
+option(onnxruntime_USE_VALGRIND "Build with valgrind hacks" OFF)
```

Comment:
Do we need to guard this from windows builds?

Reply:
Added. Thank you!